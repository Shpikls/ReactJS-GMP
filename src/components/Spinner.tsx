import React from 'react'
import { Spinner as SpinnerBootstrap } from 'react-bootstrap'
import styled from 'styled-components'

export const Spinner = (): JSX.Element => {
	return (
		<Center>
			<LoadingText>Loading...</LoadingText>
			<SpinnerBootstrap animation="border" variant="danger" />
		</Center>
	)
}

const LoadingText = styled.div`
	margin-bottom: 16px;
`

const Center = styled.div`
	position: fixed;
	top: 50%;
	left: 50%;
	transform: translate(-50%, -50%);
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
`
