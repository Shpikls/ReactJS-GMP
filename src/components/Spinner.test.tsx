import { render } from '@testing-library/react'
import React from 'react'
import { Spinner } from './Spinner'

describe('Testing Spinner.tsx', () => {
	test('To mach snapshot', () => {
		const { asFragment } = render(<Spinner />)
		expect(asFragment).toMatchSnapshot()
	})
})
