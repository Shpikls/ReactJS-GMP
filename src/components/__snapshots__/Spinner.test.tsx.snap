// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NotFound.tsx To mach snapshot 1`] = `[Function]`;

exports[`Testing Spinner.tsx To mach snapshot 1`] = `[Function]`;
